### ISSUE TITLE
With EnsureSemantics(), a Scrollable Widget with Internal Widget with NeverScrollableScrollPhysics Causes Jerky Scrolling

### ISSUE THREAD

**Issue by emma-roudabush-disney** [flutter/flutter#159358]
### Steps to reproduce

1. Have a web app with ensureSemantics ran
2. Have a listview with internal listview 
3. Have the internal listview have NeverScrollableScrollPhysics()
4. Run web app and scroll

### Expected results

Scroll is smooth

### Actual results

Scroll jerks around/jumps when scrolling up or down

If ensureSemantics is removed or if GridView or ListView (with NeverScrollableScrollPhysics) is removed or replaced, scroll is smooth.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();
  SemanticsBinding.instance.ensureSemantics();
  runApp(const TestApp());
}

class TestApp extends StatelessWidget {
  const TestApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: Scaffold(
        body: TestPage(),
      ),
    );
  }
}

class TestPage extends StatelessWidget {
  const TestPage({super.key});

  @override
  Widget build(BuildContext context) {
    return ListView(
      padding: EdgeInsets.zero,
      children: [
        SizedBox(
          height: 500.0,
          width: 400.0,
          child: ListView(
            scrollDirection: Axis.horizontal,
            physics: const NeverScrollableScrollPhysics(),
            children: [
              Container(color: Colors.red, height: 500, width: 100.0),
              Container(color: Colors.blue, height: 500, width: 100.0),
              Container(color: Colors.green, height: 500, width: 100.0),
            ],
          ),
        ),
        ...List.generate(
          20,
          (index) => ListTile(
            title: Text('Test Item $index'),
            leading: const Icon(Icons.list),
            trailing: const Icon(Icons.arrow_forward),
          ),
        ),
      ],
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/b9a318dc-1b36-4c36-9bd1-df3468f67b82

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

Was able to repro on Flutter version 3.24.2 and 3.27.0-0.1.pre
Unable to repro on Flutter version 3.22.2

</details>

**Comment by darshankawar** [comment]
@emma-roudabush-disney Thanks for the report.

I was able to replicate this on latest master, but not in latest stable (3.24.5).

1. Master : 3.27.0-1.0.pre.628


https://github.com/user-attachments/assets/7e7a437c-d1d6-48e9-9ef5-2143f9f278ee



2. stable : 3.24.5


https://github.com/user-attachments/assets/278fd940-2be7-4325-b599-f5eeac34f837

**Comment by Piinks** [comment]
Refactoring to avoid nesting ListViews is an available work around here while we investigate this issue:

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();
  SemanticsBinding.instance.ensureSemantics();
  runApp(const TestApp());
}

class TestApp extends StatelessWidget {
  const TestApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: Scaffold(
        body: TestPage(),
      ),
    );
  }
}

class TestPage extends StatelessWidget {
  const TestPage({super.key});

  @override
  Widget build(BuildContext context) {
    return CustomScrollView(
      slivers: [
        SliverToBoxAdapter(
          child: Row(
            children: [
              Container(color: Colors.red, height: 500, width: 100.0),
              Container(color: Colors.blue, height: 500, width: 100.0),
              Container(color: Colors.green, height: 500, width: 100.0),
            ],
          ),
        ),
        SliverList.list(
          children: List.generate(
            20,
            (index) => ListTile(
              title: Text('Test Item $index'),
              leading: const Icon(Icons.list),
              trailing: const Icon(Icons.arrow_forward),
            ),
          ),
        ),
      ],
    );
  }
}

```

**Comment by iapicca** [comment]
I understand a workaround on a bug, but on a regression...

**Comment by tauu** [comment]
As also reported in #159680 the scrolling issue seems to be triggered by any internal widget, that could be scrollable (as e.g. a TextField).

An initial investigation found that these internal Widgets get an node added to Semantics Tree and the Browser tiggers event for that. See also:
https://github.com/flutter/flutter/issues/159680#issuecomment-2514480179

Another general workaround, that only requires minimal code changes, is to wrap the inner Widget in a GestureDetector, with two empty callbacks. That has no functional effect on the app, but changes the generated Semantics tree. See also:
https://github.com/flutter/flutter/issues/159680#issuecomment-2515896591

**Comment by Piinks** [comment]
I think I bisected this to https://github.com/flutter/engine/pull/53922, sending to the web team to evaluate.
There are some workarounds currently available, see above.
